topics/dev/tutorials/tool-from-scratch/tutorial.md

```diff
@@ -0,0 +1,1182 @@
+> github, updates can be automated with the variables we defined in the second step, while the SHA-256 checksum ensures
+> that conda-build is getting the right source code every time.
+>
+>     {% raw %}
+>
+>     ```yaml
+>     source:
+>       url: https://github.com/davebx/{{ name }}/archive/{{ version }}.tar.gz
+>       sha256: {{ sha256 }}
+>     ```
+>
+>     {% endraw %}
+>
+> 5. Next, we move on to the build metadata. Since this is the first version of the conda recipe, the build number is 0.
+> We use the externally defined `{{ PYTHON }}` variable, which defines which python conda-build is using, to install it
+> to the build prefix. The --no-deps and --ignore-installed flags are needed to ensure that conda-build only packages
+> bellerophon itself. In this section, if necessary, we can also define patches that should be applied to the source code,
+> with the `patches:` token under `build:`, and specify that the package should not be built for a given architecture with
+> the `skip:` directive.
+>
+>    ```yaml
+>    build:
+>      noarch: python
+>      number: 0
+>      script: {{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv
+>      #patches: /dev/null # Not used in this tutorial
+>      #skip: True [osx] # Not used in this tutorial
+>    ```
+>
+> 6. After the build metadata has been defined, we need to specify dependencies for at least building and running, with
+> build-time dependencies specified in the `host:` section, and runtime dependencies in the `run:` section. With
+> bellerophon, we know that it's a python package that uses pysam to operate on SAM/BAM files, so the only runtime
+> dependencies we need are python and pysam.
+>
+>    ```yaml
+>    requirements:
+>      host:
+>        - python
+>        - pip
+>      run:
+>        - python
+>        - pysam
+>    ```
+>
+> 7. No recipe is complete without tests, and this recipe is no exception. Normally, it's sufficient to confirm that the
+> program actually runs, e.g. with a `--version` command. Some software also has a self-test flag or parameter, though
+> bellerophon is not among them, and we could even define a test script that uses test data either from the recipe or from
+> the source archive.
+>
+>    ```yaml
+>    test:
+>      commands:
+>        - bellerophon --version
+>    ```
+>
+> 8. Finally, we add information about the software, such as code's is license type, the program's
+> homepage, and optionally the github username of the person responsible for maintaining the recipe.
+>
+>    ```yaml
+>    about:
+>      home: https://github.com/davebx/bellerophon/
+>      license: MIT
+>      license_file: LICENSE
+>      summary: "Filter reads that span a mapping junction, retaining the 5'-side."
+>      maintainer: davebx # Optional
+>    ```
+>
+> Putting all these parts together, we end up with a complete conda recipe for version 1.0 of bellerophon.
+>
+> > ### {% icon question %} Question
+> >
+> > What does your final file look like?
```

Comment:
Since the parts of the file were all above, I've put this in a question to encourage learners to do the task first before they know they can just compare/see the final one.